eng/targetingpacks.targets

```diff
@@ -58,7 +58,7 @@
                          ILCompilerPackNamePattern="runtime.**RID**.Microsoft.DotNet.ILCompiler"
                          TargetFramework="$(NetCoreAppCurrent)"
                          ILCompilerPackVersion="$(ProductVersion)"
-                         ILCompilerRuntimeIdentifiers="linux-musl-x64;linux-x64;win-x64;linux-arm;linux-arm64;linux-musl-arm;linux-musl-arm64;osx-arm64;osx-x64;win-arm;win-arm64;win-x86"
+                         ILCompilerRuntimeIdentifiers="linux-musl-x64;linux-x64;win-x64;linux-arm;linux-arm64;linux-musl-arm;linux-musl-arm64;osx-arm64;osx-x64;win-arm;win-arm64;win-x86;maccatalyst-x64;maccatalyst-arm64;ios-arm64;iossimulator-arm64;iossimulator-x64;tvos-arm64;tvossimulator-arm64;tvossimulator-x64"
```

Comment:
How is it going to work with https://github.com/dotnet/runtime/pull/85047?

I was not expecting to have ILCompiler package for iOS and friends. There is not going to ILCompiler running on device. We should only need the runtime pack for the iOS and friend.

Reply:
Right, it is not necessary since we have normal runtime packs now, and ILCompiler is not needed on a device. @akoeplinger will create a PR for the Installer/SDK and close this PR.

The idea here was mostly to retrieve feedback and collect missing changes.